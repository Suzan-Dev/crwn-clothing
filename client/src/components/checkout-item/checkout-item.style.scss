.checkout-item {
  width: 100%;
  display: flex;
  min-height: 100px;
  border-bottom: 1px solid darkgrey;
  padding: 15px 0;
  font-size: 20px;
  align-items: center;

  .image-container {
    width: 23%;
    padding-right: 15px;

    img {
      width: 100%;
      height: 100%;
    }
  }
  .name,
  .quantity,
  .price {
    width: 23%;
  }

  .price {
    padding-left: 15px;
  }

  .name {
    font-size: 16px;
  }

  @media screen and (min-width: 800px) {
    .name {
      font-size: 18px;
    }
  }

  .quantity {
    display: flex;
    align-items: center;

    .arrow {
      cursor: pointer;
      padding: 5px;

      &::selection {
        background-color: transparent;
      }
    }

    .value {
      margin: 0 5px;
    }
  }

  .remove-btn {
    padding-left: 12px;
    cursor: pointer;
  }
}
