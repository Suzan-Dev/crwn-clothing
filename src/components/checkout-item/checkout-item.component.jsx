import React from 'react';
import { connect } from 'react-redux';

import './checkout-item.style.scss';

import {
  removeItemFromCart,
  addItem,
  removeItem,
} from '../../redux/cart/cart.action';

const CheckoutItem = ({ cartItem, removeItem, increaseItem, decreaseItem }) => {
  const { name, price, imageUrl, quantity } = cartItem;
  return (
    <div className='checkout-item'>
      <div className='image-container'>
        <img src={imageUrl} alt='item' />
      </div>
      <span className='name'>{name}</span>
      <span className='quantity'>
        <div className='arrow' onClick={() => decreaseItem(cartItem)}>
          &#10094;
        </div>
        <span className='value'>{quantity}</span>
        <div className='arrow' onClick={() => increaseItem(cartItem)}>
          &#10095;
        </div>
      </span>
      <span className='price'>{price}</span>
      <span className='remove-btn' onClick={() => removeItem(cartItem)}>
        &#10005;
      </span>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  removeItem: (item) => dispatch(removeItemFromCart(item)),
  increaseItem: (item) => dispatch(addItem(item)),
  decreaseItem: (item) => dispatch(removeItem(item)),
});

export default connect(null, mapDispatchToProps)(CheckoutItem);
