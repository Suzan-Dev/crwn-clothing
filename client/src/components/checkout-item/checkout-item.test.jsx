import React from 'react';
import { shallow } from '../../setupTests';
import { CheckoutItem } from './checkout-item.component';

describe('Testing: CheckoutItem component!', () => {
  const mockCartItem = {
    imageUrl: 'www.testImage.com',
    price: 9,
    name: 'PS5',
    quantity: 2,
  };
  const mockRemoveItem = jest.fn();
  const mockIncreaseItem = jest.fn();
  const mockDecreaseItem = jest.fn();

  const mockProps = {
    cartItem: mockCartItem,
    removeItem: mockRemoveItem,
    increaseItem: mockIncreaseItem,
    decreaseItem: mockDecreaseItem,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  test('should render CheckoutItem component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should dispatch decreaseItem action when left arrow is clicked.', () => {
    wrapper.find('.quantity').childAt(0).simulate('click');
    expect(mockDecreaseItem).toHaveBeenCalledWith(mockCartItem);
  });

  test('should dispatch increaseItem action when right arrow is clicked.', () => {
    wrapper.find('.quantity').childAt(2).simulate('click');
    expect(mockIncreaseItem).toHaveBeenCalledWith(mockCartItem);
  });

  test('should dispatch removeItem action when remove button is clicked.', () => {
    wrapper.find('.remove-btn').simulate('click');
    expect(mockRemoveItem).toHaveBeenCalledWith(mockCartItem);
  });
});
