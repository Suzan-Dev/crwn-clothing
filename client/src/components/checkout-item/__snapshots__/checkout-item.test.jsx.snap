// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: CheckoutItem component! should render CheckoutItem component. 1`] = `
"<div className=\\"checkout-item\\">
  <div className=\\"image-container\\">
    <img src=\\"www.testImage.com\\" alt=\\"item\\" />
  </div>
  <span className=\\"name\\">
    PS5
  </span>
  <span className=\\"quantity\\">
    <div className=\\"arrow\\" onClick={[Function: onClick]}>
      ❮
    </div>
    <span className=\\"value\\">
      2
    </span>
    <div className=\\"arrow\\" onClick={[Function: onClick]}>
      ❯
    </div>
  </span>
  <span className=\\"price\\">
    9
  </span>
  <span className=\\"remove-btn\\" onClick={[Function: onClick]}>
    ✕
  </span>
</div>"
`;
